arc smp build failed with mwdt toolchain.

**Describe the bug**
arc smp build failed with mwdt toolchain.

**To Reproduce**
Steps to reproduce the behavior:
1. west build -b nsim_hs_smp -d build samples/hello_world/
2. See error


**Logs and console output**
```
-- west build: generating a build system
Including boilerplate (Zephyr base): /SCRATCH2/Watson/zephyrproject/zephyr/cmake/app/boilerplate.cmake
-- Application: /SCRATCH2/Watson/zephyrproject/zephyr/samples/hello_world
-- Zephyr version: 2.6.99 (/SCRATCH2/Watson/zephyrproject/zephyr), build: zephyr-v2.6.0-1858-ga9779bb2dcd3
-- Found Python3: /depot/Python/Python-3.7.0/bin/python3.7 (found suitable exact version "3.7.0") found components: Interpreter 
-- Found west (found suitable version "0.11.0", minimum required is "0.7.1")
-- Board: nsim_hs_smp
-- Cache files will be written to: /u/zhiwei/.cache/zephyr
-- Using toolchain: zephyr 0.12.3 (/SCRATCH2/svc-arcoss_auto/tools/zephyr-sdk/0.12.3)
-- Found dtc: /SCRATCH2/svc-arcoss_auto/tools/zephyr-sdk/0.12.3/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.5.0", minimum required is "1.4.6")
-- Found BOARD.dts: /SCRATCH2/Watson/zephyrproject/zephyr/boards/arc/nsim/nsim_hs_smp.dts
node '/iccm@0' compatible 'arc,iccm' has unknown vendor prefix 'arc'
node '/dccm@80000000' compatible 'arc,dccm' has unknown vendor prefix 'arc'
-- Generated zephyr.dts: /SCRATCH2/Watson/zephyrproject/zephyr/build/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: /SCRATCH2/Watson/zephyrproject/zephyr/build/zephyr/include/generated/devicetree_unfixed.h
-- Generated device_extern.h: /SCRATCH2/Watson/zephyrproject/zephyr/build/zephyr/include/generated/device_extern.h
Parsing /SCRATCH2/Watson/zephyrproject/zephyr/Kconfig
Loaded configuration '/SCRATCH2/Watson/zephyrproject/zephyr/boards/arc/nsim/nsim_hs_smp_defconfig'
Merged configuration '/SCRATCH2/Watson/zephyrproject/zephyr/samples/hello_world/prj.conf'
Configuration saved to '/SCRATCH2/Watson/zephyrproject/zephyr/build/zephyr/.config'
Kconfig header saved to '/SCRATCH2/Watson/zephyrproject/zephyr/build/zephyr/include/generated/autoconf.h'
-- The C compiler identification is Clang 11.1.0
-- The CXX compiler identification is Clang 11.1.0
-- The ASM compiler identification is unknown
-- Found assembler: /global/apps/mwdt_2021.06/MetaWare/arc/bin/ccac
-- Warning: Did not find file Compiler/-ASM
CMake Warning at ../../cmake/linker/arcmwdt/target.cmake:138 (message):
  MWDT toolchain does not support

             CONFIG_LINKER_ORPHAN_SECTION_WARN
Call Stack (most recent call first):
  ../../CMakeLists.txt:250 (toolchain_ld_baremetal)


-- Configuring done
-- Generating done
-- Build files have been written to: /SCRATCH2/Watson/zephyrproject/zephyr/build
-- west build: building application
[1/120] Preparing syscall dependency handling

[113/120] Linking C executable zephyr/zephyr_prebuilt.elf
FAILED: zephyr/zephyr_prebuilt.elf zephyr/zephyr_prebuilt.map 
: && /global/apps/mwdt_2021.06/MetaWare/arc/bin/ccac    zephyr/CMakeFiles/zephyr_prebuilt.dir/misc/empty_file.c.obj  -o zephyr/zephyr_prebuilt.elf  -Wl,-T/SCRATCH2/Watson/zephyrproject/zephyr/build/zephyr/linker_zephyr_prebuilt.cmd  -Wl,--gc-sections  -Wl,--entry=__start  -Wl,--Map=/SCRATCH2/Watson/zephyrproject/zephyr/build/zephyr/zephyr_prebuilt.map  -Wl,--whole-archive  app/libapp.a  zephyr/libzephyr.a  zephyr/arch/common/libarch__common.a  zephyr/arch/arch/arc/core/libarch__arc__core.a  zephyr/lib/libc/minimal/liblib__libc__minimal.a  zephyr/lib/posix/liblib__posix.a  zephyr/drivers/serial/libdrivers__serial.a  -Wl,--no-whole-archive  zephyr/kernel/libkernel.a  zephyr/CMakeFiles/offsets.dir/./arch/arc/core/offsets/offsets.c.obj  -L/SCRATCH2/Watson/zephyrproject/zephyr/build/zephyr  zephyr/arch/common/libisr_tables.a  -Wl,--sort-section=alignment  -Wl,-u_OffsetAbsSyms  -Wl,-u_ConfigAbsSyms  -Hlld  -Hnosdata  -Hnocrt  -Xtimer0  -Hnoxcheck_obj  -Hnocplus  -Hheap=0  -Hnoivt  -Hlib=hs38_full && cd /SCRATCH2/Watson/zephyrproject/zephyr/build/zephyr && /depotbld/RHEL7.0/Python/Python-3.7.0/lib/python3.7/site-packages/cmake/data/bin/cmake -E echo
lldac: error: undefined symbol: @_curr_cpu
>>> referenced by fault_s.S:89 (/SCRATCH2/Watson/zephyrproject/zephyr/arch/arc/core/fault_s.S:89)
>>>               fault_s.S.obj:(.text.__fault+0x14) in archive zephyr/arch/arch/arc/core/libarch__arc__core.a
>>> referenced by regular_irq.S:222 (/SCRATCH2/Watson/zephyrproject/zephyr/arch/arc/core/regular_irq.S:222)
>>>               regular_irq.S.obj:(.text._rirq_enter+0x40) in archive zephyr/arch/arch/arc/core/libarch__arc__core.a
>>> referenced by /tmp/4930_1fast_irq.s
>>>               fast_irq.S.obj:(.text._firq_exit+0x80) in archive zephyr/arch/arch/arc/core/libarch__arc__core.a
>>> referenced 2 more times
>>> did you mean: _curr_cpu
>>> defined in: zephyr/arch/arch/arc/core/libarch__arc__core.a(arc_smp.c.obj)
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /depot/Python/Python-3.7.0/bin/cmake --build /SCRATCH2/Watson/zephyrproject/zephyr/build
```

**Environment (please complete the following information):**
 - OS: (Linux)
 - Toolchain (arcmwdt)
 - Commit SHA (a9779bb2dcd372788a6aff2982d460e1be927aaf)


**Additional context**
this failure was introduced by this commit: 7df1fc7cbc364c7209395251d7d242fce626997b

--- a/arch/arc/include/swap_macros.h
+++ b/arch/arc/include/swap_macros.h
@@ -386,7 +386,7 @@
 	/* get pointer to _cpu_t of this CPU */
 	_get_cpu_id MACRO_ARG(irq_sp)
 	ASLR MACRO_ARG(irq_sp), MACRO_ARG(irq_sp), ARC_REGSHIFT
-	LDR MACRO_ARG(irq_sp), MACRO_ARG(irq_sp), @_curr_cpu
+	LDR MACRO_ARG(irq_sp), MACRO_ARG(irq_sp), _curr_cpu
 	/* get pointer to irq_stack itself */
 	LDR MACRO_ARG(irq_sp), MACRO_ARG(irq_sp), ___cpu_t_irq_stack_OFFSET
 #else

